lib/std.dl

```diff
@@ -77,29 +77,50 @@ extern function str_to_lower(s: string): string
 extern function hash64(x: 'X): bit<64>
 extern function hash128(x: 'X): bit<128>
 
+/* The `Group` type is used exclusively in aggregation operations.  It
+ * represents a non-empty list of objects sorted in ascending order.
+ */ 
+extern type Group<'A>
+
 /*
  * Standard aggregates
  */
 
-extern type Group<'A>
+/* Returns the number of elements in the group, which is guaranteed
+ * to be >0. */
```

Comment:
Here you are assuming that groups will be only produced by aggregation.
However, if you allow functions on groups perhaps they can be produced in some other ways as well.
So I don't think you should promise that a group cannot be empty.

Reply:
Groups are immutable and can only be produced by the `Aggregate` operator, so there is no way for the programmer to forge an empty group.  In fact, I should not even allow copying groups or returning them from functions, as this will generate code that will fail to compile in Rust, but I don't have a clean infrastructure to do dataflow analysis atm.